test/interchain/inter_chain_accounts_test.go

```diff
@@ -0,0 +1,119 @@
+// an ICA on Celestia (host chain) using the Cosmos Hub (controller chain).
+func TestInterChainAccounts(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping TestInterChainAccounts in short mode.")
+	}
+
+	client, network := interchaintest.DockerSetup(t)
+	celestia := chainspec.GetCelestia(t)
+	cosmosHub := chainspec.GetCosmosHub(t)
+	relayer := getRelayerFactory(t).Build(t, client, network)
+	pathName := fmt.Sprintf("%s-to-%s", celestia.Config().ChainID, cosmosHub.Config().ChainID)
+	interchain := interchaintest.NewInterchain().
+		AddChain(celestia).
+		AddChain(cosmosHub).
+		AddRelayer(relayer, getRelayerName()).
+		AddLink(interchaintest.InterchainLink{
+			Chain1:  celestia,
+			Chain2:  cosmosHub,
+			Relayer: relayer,
+			Path:    pathName,
+		})
+
+	ctx := context.Background()
+	reporter := testreporter.NewNopReporter().RelayerExecReporter(t)
+	err := interchain.Build(ctx, reporter, interchaintest.InterchainBuildOptions{
+		TestName:  t.Name(),
+		Client:    client,
+		NetworkID: network,
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() { _ = interchain.Close() })
+
+	err = relayer.StartRelayer(ctx, reporter, pathName)
+	require.NoError(t, err)
+
+	err = testutil.WaitForBlocks(ctx, 2, celestia, cosmosHub)
+	require.NoError(t, err)
+
+	celestiaConnections, err := relayer.GetConnections(ctx, reporter, celestia.Config().ChainID)
+	require.NoError(t, err)
+	require.Len(t, celestiaConnections, 1)
+
+	cosmosConnections, err := relayer.GetConnections(ctx, reporter, cosmosHub.Config().ChainID)
+	require.NoError(t, err)
+	require.Len(t, cosmosConnections, 2) // 2 connections: the first is connection-0 and the second is connection-localhost.
+	cosmosConnection := cosmosConnections[0]
+
+	users := interchaintest.GetAndFundTestUsers(t, ctx, t.Name(), math.NewInt(10_000_000_000), celestia, cosmosHub)
+	err = testutil.WaitForBlocks(ctx, 2, celestia, cosmosHub)
+	require.NoError(t, err)
+
+	celestiaUser, cosmosUser := users[0], users[1]
+	celestiaAddr := celestiaUser.(*cosmos.CosmosWallet).FormattedAddressWithPrefix(celestia.Config().Bech32Prefix)
+	cosmosAddr := cosmosUser.(*cosmos.CosmosWallet).FormattedAddressWithPrefix(cosmosHub.Config().Bech32Prefix)
+	fmt.Printf("celestiaAddr: %s, cosmosAddr: %v\n", celestiaAddr, cosmosAddr)
+
+	registerICA := []string{
+		cosmosHub.Config().Bin, "tx", "interchain-accounts", "controller", "register", cosmosConnection.ID,
+		"--chain-id", cosmosHub.Config().ChainID,
+		"--home", cosmosHub.HomeDir(),
+		"--node", cosmosHub.GetRPCAddress(),
+		"--from", cosmosUser.KeyName(),
+		"--keyring-backend", keyring.BackendTest,
+		"--fees", fmt.Sprintf("300000%v", cosmosHub.Config().Denom),
+		"--gas", "300000", // the auto gas estimation underestimates the gas required.
+		"--yes",
+	}
+	stdout, stderr, err := cosmosHub.Exec(ctx, registerICA, nil)
+	require.NoError(t, err)
+	require.Empty(t, stderr)
+	t.Logf("stdout %v", string(stdout))
+
+	err = testutil.WaitForBlocks(ctx, 5, celestia, cosmosHub)
```

Comment:
```suggestion
	err = testutil.WaitForBlocks(ctx, 1, celestia, cosmosHub)
```
Can this just be 1? Is the timeout for celestia going to be 11 seconds? i.e. the default

Reply:
I just dropped it to `1` and hit an error because 

```
    inter_chain_accounts_test.go:103:
        	Error Trace:	/Users/rootulp/git/rootulp/celestiaorg/celestia-app/test/interchain/inter_chain_accounts_test.go:103
        	Error:      	Received unexpected error:
        	            	exit code 1:  Error: rpc error: code = NotFound desc = rpc error: code = NotFound desc = failed to retrieve account address for icacontroller-cosmos15c8ytdk94gw4ksajeekzlmx6qjrnu4ul20l094 on connection connection-0: key not found
```

blocks are getting produced much faster than 11 seconds b/c the test only takes ~80 seconds to complete and we're at least waiting for 9 blocks in the whole test (2, 2, then 5 here):

```
ok  	github.com/celestiaorg/celestia-app/test/interchain	82.041s
```